### PR TITLE
feat: subscribe to finalized head, no encoding for username

### PR DESCRIPTION
**Pull request by driemworks**

### CODE REVIEW

File: src/MurmurClient.ts [hunk 1]
```diff
@@ -82,7 +88,7 @@ export class MurmurClient {
     }
     const request: NewRequest = {
       validity,
-      current_block: (await this.getCurrentBlock()).toNumber(),
+      current_block: this.finalizedBlockNumber,
```
Comment (juangirini) [inline src/MurmurClient.ts:91]: can't this value be 0? if so we should implement an async function that await for this value to be > 0

File: src/MurmurClient.ts [hunk 2]
```diff
@@ -169,12 +169,12 @@ export class MurmurClient {
         console.log(
           `Transaction included at blockHash ${result.status.asInBlock}`
         );
+        callback(result);
```
Comment (juangirini) [inline src/MurmurClient.ts:172]: why do we want the callback before the transaction is finalized?
Should we have two callbacks?

**Review by juangirini — COMMENTED** [review]
Please see my comments, I am unsure about `finalizedBlockNumber` and the `callback`

File: src/MurmurClient.ts (hunk 2 above)
Comment (driemworks) [reply src/MurmurClient.ts:172]: We could have two callbacks, but the reason I moved it is UX. Waiting for it to be finalized just takes longer and being in a block is good enough for the demo imo.

Comment (juangirini) [reply src/MurmurClient.ts:172]: ok fair enough, let's leave it there for now

**Pull request merged**